client/src/pages/Profile.js

```diff
@@ -0,0 +1,96 @@
+import React, {Component} from 'react';
+import ProjectCard from '../components/projectCard';
+import { withStyles, Button, Grid } from '@material-ui/core';
+import { getUser } from '../api/users';
+
+class ProfilePage extends Component {
+  state = {
+    profile: {
+      id: this.props.match.params.id,
+      name: '',
+      location: '',
+      about: '',
+      expertise: '',
+      projects: [],
+      imageUrl: ''
+    },
+    currentUserId: localStorage.getItem('id')
+  };
+
+  componentDidMount() {
+    const { id } = this.state.profile;
+    const profile = getUser(id).then((profile) => {
+      this.setState({ profile: profile.data });
+    });
+  }
+
+  renderUserInfo() {
+    const { imageUrl, name, location, about, expertise } = this.state.profile;
+
+    return (
+      <div className="profile-details">
+        <div>
+            <div><h3>{name}</h3></div>
+            <div><p>location</p></div>
+        </div>
+        {this.renderButton()}
+        <div>
+            <div><p>about me</p></div>
+            <div><p>expertise</p></div>
+        </div>
+      </div>
+    );
+  }
+
+  renderButton() {
+    const { profile, currentUserId } = this.state;
+    const buttonType = profile.id === currentUserId ? 'edit' : 'message';
+
+    return (
+      <Button type="submit" variant="outlined" color="primary">
+        {buttonType}
+      </Button>
+    );
+  }
```

Comment:
Is there a reason you used a function to render button instead of just putting the following under the profile details div?
Also you could put the destructing before the return in the renderUserInfo return
(button does not need to be submit, type="submit" is html for form buttons)
```
<Button variant="outlined" color="primary" onClick={this.handleClick}>
        { profile.id === currentUserId ? 'edit' : 'message' }
</Button>
```

Reply:
Just process, really. I wasn't sure what kind of logic would be involved in the conditional render of the button. I usually start by keeping it separate until it's settled, then I refactor.